### PR TITLE
feat: move state objects from autora-workflow

### PR DESCRIPTION
**Pull request by hollandjg**
move the state objects from autora-workflow: autora.state to the core package

depends on https://github.com/AutoResearch/autora-workflow/pull/28

### CODE REVIEW

File: docs/cycle/Linear and Cyclical Workflows using Functions and States.ipynb [hunk 1]
```diff
@@ -0,0 +1,1108 @@
...
+   "metadata": {},
+   "source": [
+    "# Linear And Cyclical Workflows Using Functions And States"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Using the functions in `autora.workflow`, we can build flexible pipelines and cycles which operate on state objects.\n"
```
Comment (younesStrittmatter) [inline docs/cycle/Linear and Cyclical Workflows using Functions and States.ipynb:14]:
```suggestion
"Using the functions in `autora.state`, we can build flexible pipelines and cycles which operate on state objects.\n"
```

File: docs/cycle/Linear and Cyclical Workflows using Functions and States.ipynb [hunk 2]
```diff
@@ -0,0 +1,1108 @@
...
+    "We define a two part AER pipeline consisting of an experiment runner and a theorist (we use the seed conditions\n",
+    "always).\n",
+    "\n",
+    "The key part here is that both experiment runner and theorist are functions which:\n",
+    "- operate on the `State`, and\n",
+    "- return a modified object of the **same type** `State`.\n",
+    "\n",
+    "### Defining The State\n",
+    "\n",
+    "We define the state as a dataclass, subclassed from `autora.workflow.State` with fields representing the variables,\n",
```
Comment (younesStrittmatter) [inline docs/cycle/Linear and Cyclical Workflows using Functions and States.ipynb:59]:
```suggestion
    "We define the state as a dataclass, subclassed from `autora.state.delta.State` with fields representing the variables,\n",
```

File: docs/cycle/Linear and Cyclical Workflows using Functions and States.ipynb [hunk 3]
```diff
@@ -0,0 +1,1108 @@
...
+    "parameters, experimental data, (possibly) conditions, and (possibly) a model.\n",
+    "\n",
+    "This state has no \"history\"; it represents a snapshot of the data at one time. Other exemplar state objects are\n",
+    "available in the subpackage `autora.workflow.state` and include some with in-built histories."
```
Comment (younesStrittmatter) [inline docs/cycle/Linear and Cyclical Workflows using Functions and States.ipynb:63]:
```suggestion
    "available in the subpackage `autora.state` and include some with in-built histories."
```

**Review by younesStrittmatter — CHANGES_REQUESTED** [review]
almost perfect.

**Review by benwandrew — APPROVED** [review]
removed an empty cell in Linear and Cyclical Workflows using Functions and States.ipynb; otherwise, looks good to move onto a better place

**Review by younesStrittmatter — APPROVED** [review]
looks good

**Pull request merged**